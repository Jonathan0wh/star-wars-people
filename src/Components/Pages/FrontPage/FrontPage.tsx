import { Col, Divider, Layout, Row } from "antd";
import PeopleTable from "Components/Views/PeopleTable";
import PersonDetail from "Components/Views/PersonDetail";

function FrontPage() {
  const { Content } = Layout;

  return (
    <Layout>
      <Content>
        <Row justify="center" align="middle">
          <Col>
            <PeopleTable />
            <Divider />
            <PersonDetail />
          </Col>
        </Row>
      </Content>
    </Layout>
  );
}

export default FrontPage;
