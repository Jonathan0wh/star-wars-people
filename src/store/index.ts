export { store, persistor } from "./store";

export type { RootState, AppDispatch } from "./store";
