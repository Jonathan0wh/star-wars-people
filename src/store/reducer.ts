import { combineReducers } from "redux";
import { swApi } from "services/swApi";

export const rootReducer = combineReducers({
  [swApi.reducerPath]: swApi.reducer,
});
