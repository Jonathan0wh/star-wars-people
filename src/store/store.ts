import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/query";
import { personDetailReducer } from "Components/Views/PersonDetail";
import { swApi } from "services/swApi";

export const store = configureStore({
  reducer: {
    [swApi.reducerPath]: swApi.reducer,

    personDetail: personDetailReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(swApi.middleware),
});

setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
