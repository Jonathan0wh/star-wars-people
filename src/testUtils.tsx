// test-utils.js
import React from "react";
import { applyMiddleware, createStore } from "redux";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { render as rtlRender, RenderOptions } from "@testing-library/react";
// Import your own reducer
import { rootReducer } from "store/reducer";

type renderProps = {
  initialState?: object;
  store?: ReturnType<typeof createStore>;
} & RenderOptions;

function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  }: renderProps = {}
) {
  return rtlRender(ui, {
    wrapper: ({ children }: { children?: React.ReactNode }) => (
      <Provider store={store}>{children}</Provider>
    ),
    ...renderOptions,
  });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
