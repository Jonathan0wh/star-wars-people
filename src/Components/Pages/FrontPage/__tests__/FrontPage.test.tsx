import { store } from "store";
import { render } from "testUtils";

import FrontPage from "../FrontPage";

describe("Page Component: Front Page", () => {
  it("should render by default", () => {
    const component = render(<FrontPage />);

    const tree = component.asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render with correct store data", () => {
    const component = render(<FrontPage />, { store });

    const tree = component.asFragment();
    expect(tree).toMatchSnapshot();
  });
});
