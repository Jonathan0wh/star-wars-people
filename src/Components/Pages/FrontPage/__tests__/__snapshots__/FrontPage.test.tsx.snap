// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component: Front Page should render by default 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
    style="min-height: 100vh;"
  >
    <main
      class="ant-layout-content"
    >
      <div
        class="ant-row ant-row-center ant-row-middle"
      >
        <div
          class="ant-col"
        >
          <div
            class="ant-row ant-row-center"
          >
            <div
              class="ant-col column_container"
            >
              <div
                class="ant-skeleton ant-skeleton-active skeleton"
              >
                <div
                  class="ant-skeleton-content"
                >
                  <h3
                    class="ant-skeleton-title"
                    style="width: 200px;"
                  />
                  <ul
                    class="ant-skeleton-paragraph"
                  >
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li
                      style="width: 100%;"
                    />
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-divider ant-divider-horizontal"
            role="separator"
          />
          <div
            class="ant-row row_container"
          >
            <div
              class="ant-col"
            />
          </div>
        </div>
      </div>
    </main>
  </section>
</DocumentFragment>
`;

exports[`Page Component: Front Page should render with correct store data 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
    style="min-height: 100vh;"
  >
    <main
      class="ant-layout-content"
    >
      <div
        class="ant-row ant-row-center ant-row-middle"
      >
        <div
          class="ant-col"
        >
          <div
            class="ant-row ant-row-center"
          >
            <div
              class="ant-col column_container"
            >
              <div
                class="ant-skeleton ant-skeleton-active skeleton"
              >
                <div
                  class="ant-skeleton-content"
                >
                  <h3
                    class="ant-skeleton-title"
                    style="width: 200px;"
                  />
                  <ul
                    class="ant-skeleton-paragraph"
                  >
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li />
                    <li
                      style="width: 100%;"
                    />
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-divider ant-divider-horizontal"
            role="separator"
          />
          <div
            class="ant-row row_container"
          >
            <div
              class="ant-col"
            />
          </div>
        </div>
      </div>
    </main>
  </section>
</DocumentFragment>
`;
